rfd/0051-x11-forwarding.md

```diff
@@ -0,0 +1,205 @@
+and the local X Server. However, the X authorization data in the `x11` channel request
+was the fake data we created. So before forwarding the X request to the X Server, the
+SSH client replaces the fake data with real X authorization data. This data will either
+be trusted xauth data, or untrusted xauth data, depending on what the client requested.
+
+##### Untrusted X11 Forwarding
+
+Untrusted X authorization data must be generated through `xauth` in order to be recognized
+by the X Server. To do this, the client runs `xauth generate <$DISPLAY> untrusted <timeout?>`.
+Any X Client request using this untrusted token will be be subject to the X Security extension,
+granting it fewer privileges than usual. 
+
+##### Trusted X11 Forwarding
+
+In trusted X11 forwarding, we can rely on the client's local environment. If we fill in 
+the X authorization packet in the request with a random cookie, the X Server will just
+ignore it and connect as if it received the request locally.
+
+### tsh UX
+
+`tsh ssh -X` or `tsh ssh -Y` can be used to request a session with `trusted` X11 forwarding.
+However, if `tsh ssh -X` is used with the `-oForwardX11Trusted=yes` flag, it will instead
+request `untrusted` forwarding, while `tsh ssh -Y` would still request `trusted`.
+
+This may seem unintuitive, and it is, but this is the same UX as OpenSSH client. In OpenSSH,
+to use untrusted X11 forwarding you have to set the `ForwardX11Trusted no` in the client's
+ssh config, so we simply mimic that behavior with our existing OpenSSH flag options.
+
+The reason for this oddity is essentially that `trusted` X11 forwarding (aka the X Security
+extension) is an opt in extension of the X Protocol with negative performance and compatibility
+implications.
+
+### X Security extension
+
+The X Security extension was created about 10 years after the 11th version of X. The
+extension introduced `untrusted` tokens which could be used to grant fewer X privileges
+to the user of the token. It also introduced the ability to set a timeout for the token.
+
+In contrast, normal `trusted` tokens give full unmitigated X privileges, allowing for action
+such as screenshots, monitoring peripherals, and many other actions one would expect to
+perform on local machines. In the case of X11 forwarding however, this means that the
+client is providing the SSH server unmitigated access to its local X Server. If the server
+becomes compromised or another user has access to your remote user, then the Client's
+local XServer could be subject to attacks such as keystroke monitoring to capture a password.
+
+Unfortunately, since the X Security extension was created as an afterthought, it is not
+built into the X protocol and has some serious performance drawbacks. The performance 
+drop is somewhere in the ballpark of 10x. Additionally, many X Client's don't support
+the X Security Extension properly and may crash immediately or at some point during 
```

Comment:
Can we link to any issues or examples here?

Reply:
I added a general example but couldn't find anything specific. This may be a much smaller issue than it was 20 years ago, hard to tell to be honest.